pom.xml

```diff
@@ -133,6 +133,26 @@
             </descriptorRefs>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <version>0.8.6</version>
+          <executions>
+            <execution>
+              <id>default-prepare-agent</id>
+              <goals>
+                <goal>prepare-agent</goal>
+              </goals>
+            </execution>
+            <execution>
+              <id>default-report</id>
+              <phase>test</phase>
+              <goals>
+                <goal>report</goal>
+              </goals>
+            </execution>
+          </executions>
+        </plugin>
```

Comment:
You've put this inside of the `pluginManagement` element. That's meant for sharing configuration across multiple modules (where you have this in the parent pom), but it will not actually enable the plugin for any module. To enable the plugin, you must add this inside of the `build` element:

```java
    <plugins>
      <plugin>
        <groupId>org.jacoco</groupId>
        <artifactId>jacoco-maven-plugin</artifactId>
      </plugin>
    </plugins>
```

However, since we have no intention of creating a multi-module project, you may as well just move the entire jacoco configuration into the above shown `plugin` element.

Note that configuring the default plugins inside `pluginManagement` works simply because they are enabled by default. It also works for spotless since that is explicitly invoked in a separate phase, but it does not work for plugins that bind to existing phases, as they then don't get enabled.

Reply:
> but it does not work for plugins that bind to existing phase

I was trying to invoke it as a separate phase too - `mvn jacoco:report` but it did not work. But like you said, `spotless` worked fine.

Reply:
You can invoke `jacoco:report`. It just won't do anything as the coverage files must be generated first, and that happens during the `test` phase.